content/develop/tutorials/databases/snowflake.md

```diff
@@ -7,151 +7,220 @@ slug: /develop/tutorials/databases/snowflake
+
+   </Important>
+
+1. Add your connection parameters to `.streamlit/secrets.toml`:
 
-<Tip>
+   ```toml
+   [connections.snowflake]
+   account = "xxxxxxx-xxxxxxx"
+   user = "xxx"
+   private_key_file = "../xxx/xxx.p8"
+   role = "xxx"
+   warehouse = "xxx"
+   database = "xxx"
+   schema = "xxx"
+   ```
 
-Make sure your account name is in the format `<my_organization>-<my_account>`. This is the general-purpose identifier format and not the `.`-separated format used for SQL commands.
+   <Important>
 
-</Tip>
+   Your account identifier must be hyphen-separated: `<my_organization>-<my_account>`. This is the general-purpose identifier format and not the period-separated format used within SQL statements.
 
-If you created the database from the previous step, the names of your database and schema are `PETS` and `PUBLIC`, respectively. Streamlit will also use **Snowflake config and credentials** from a [SnowSQL config file](https://docs.snowflake.com/en/user-guide/snowsql-config#snowsql-config-file) if available.
+   </Important>
 
-<Important>
+   In the example above, the connection uses key-pair authentication. Therefore, `private_key_file` is defined instead of `password`. `private_key_file` can be an absolute or relative path. If you use a relative path, it should be relative to your app's working directory (where you execute `streamlit run`).
 
-Add this file to `.gitignore` and don't commit it to your GitHub repo!
+### Option 2: Use `.snowflake/connections.toml`
 
-</Important>
+If you already have your connection configured using [Snowflake's connections file](https://docs.snowflake.com/en/developer-guide/python-connector/python-connector-connect#connecting-using-the-connections-toml-file), you can use it as-is. If you are using a default connection, no change is needed in later steps of this tutorial. If you are using a named connection, you will need to include the name in `st.connection`. This is noted in a later step. For information about using named connections, see the examples in the API reference for [`SnowflakeConnnection`](/develop/api-reference/connections/st.connections.snowflakeconnection).
+
+1. If you don't already have a `.snowflake/configuration.toml` file in your user directory, create an empty connections file.
+1. Add your connection parameters to `.snowflake/connection.toml`:
+
+   ```toml
+   [default]
+   account = "xxxxxxx-xxxxxxx"
+   user = "xxx"
+   private_key_file = "../xxx/xxx.p8"
+   role = "xxx"
+   warehouse = "xxx"
+   database = "xxx"
+   schema = "xxx"
+   ```
+
+   This example uses key-pair authentication as described in the previous option.
 
 ## Write your Streamlit app
 
-Copy the code below to your Streamlit app and run it. Make sure to adapt the query to use the name of your table.
+1. Copy the code below to your Streamlit app and save it. If you are not using the example database and table from the first section of this tutorial, replace the SQL query and results handling as desired.
```

Comment:
"the following code" instead of "the code below." 

"as appropriate" instead of "as desired." Snowflake doesn't deal with desires!